core/src/main/java/com/sequenceiq/cloudbreak/service/image/ImageCatalogService.java

```diff
@@ -121,10 +115,7 @@
     private TransactionService transactionService;
 
     @Inject
-    private PrefixMatcherService prefixMatcherService;
-
-    @Inject
-    private LatestDefaultImageUuidProvider latestDefaultImageUuidProvider;
+    private ImageCatalogServiceProxy imageCatalogServiceProxy;
```

Comment:
ImageCatalogServiceProxy injected into ImageCatalogService, sounds like it should be backwards. I think the version and advertised services are the strategies, and ImageCatalogServiceProxy is more like ImageCatalogHandler? Naming things is really hard :smile:

Reply:
Yes it is :-). We use similar naming convention in CB for the same pattern. In my point of view it is not the case where we should use "handler" naming. I think the best would be to rename the original ImageCatalogService to ImageService but it might mess veterans up :-D